### ISSUE TITLE
Errors are too prominent for mobile screens.

### ISSUE THREAD

**Issue by danielo515** [Jermolene/TiddlyWiki5#1743]
Hello,

Did someone noticed how big and annoying are error messages on mobile screens? Just wait to have 2 or 3 error messages and you have an unusable wiki. This is worst if the messages are in a loop that could not be a problem. I think some kind of "acknowledge" is needed for this kind of situations. Or at least, make errors less prominent.

**Comment by pmario** [comment]
Which error do you mean?

**Comment by danielo515** [comment]
Those "Warnings" that you have to delete to get rid of them

**Comment by Jermolene** [comment]
@danielo515 the documentation calls these "alerts":

![screenshot 2015-05-18 17 38 20](https://cloud.githubusercontent.com/assets/174761/7685388/d055dd92-fd85-11e4-9f36-7286a82b65e6.png)

I'm happy to accept suggestions of alternatives ways of working on mobile.

**Comment by danielo515** [comment]
Hello @Jermolene 

Yes, I'm talking about alerts. Honestly, I can't imagine a good approach for this, but at least a "acknowledge this" button would be ok. Once acknowledged it should not appear again for a while or until next boot (opening again the file? reloading the page?)

**Comment by pmario** [comment]
Those alerts are especially annoying, if you lost your server connection, since they pop up every minute or so.

**Comment by Jermolene** [comment]
> if you lost your server connection, since they pop up every minute or so.

The intention is that consecutive errors of the same type will get coalesced together. Is that not what's happening for you?

**Comment by danielo515** [comment]
> The intention is that consecutive errors of the same type will get coalesced together. Is that not what's happening for you?

Yes they are, but they are annoying anyway. With just one of these messages the TW is totally unusable in a mobile device. And you can not get rid of it because as @pmario said, they pop up every minute or even less in many cases.

**Comment by danielo515** [comment]
Hello @Jermolene ,

Please take a look at this screenshot. This is an high end device with 5" of screen:

![image](https://cloud.githubusercontent.com/assets/2270425/7687665/1fc580a8-fd9f-11e4-9967-e0fe6d783083.png)

**Comment by Jermolene** [comment]
I'm open to improvements. The easiest option would be to restyle the alerts; the best option would be to refactor the tiddlyweb adaptor so that instead of logging errors it maintains a simple ambient visual indication of connection status.

**Comment by danielo515** [comment]
Hello @Jermolene 

From my point of view what have to be changed is the alert system. Changing how tiddlyweb behaves will only work for tiddlyweb, and we should repeat the same work for every other adapter. What about a dismiss mechanism?

**Comment by Jermolene** [comment]
Hi @danielo515 at the moment, the syncadaptor calls the logger which displays the alert. So there are several ways that we could change this:

* The syncadaptor could do something else other than call the logger. For example, update a status tiddler that allows a UI beacon to be shown
* The logger could do something else other than display alerts
* The alerts could be formatted differently

**Comment by danielo515** [comment]
Hello @Jermolene 

What do you  mean with syncadaptor? Every particular syncadaptor or the general syncer of TW?

I like your proposals, but what about this one?:

* Add a button to the alert to acknowledge it
* The alert title is added to a state tiddler list (to avoid save or sync of this information)
* The warning is only showed if its title is not in the dismiss notification list.

How that sounds?

**Comment by Jermolene** [comment]
Hi @danielo515 I meant that the tiddlywebadaptor could do something else other than call the logger.

> I like your proposals, but what about this one?:

You are proposing a lot more complexity in the alert mechanism. I think the first option is better: for the sync adaptor to stop using alerts for this purpose.

By the way, I meant to ask you more about the situations in which you'd want to continue working after losing server contact. Is it a matter of wanting to continue with read-only access to a wiki even after you've lost the connection?

**Comment by pmario** [comment]
I think, he just wants to continue and store the tiddlers to the local pouchdb. ... If connection comes back, a sync can be done.

**Comment by danielo515** [comment]
Hello @Jermolene, as you have said, what I want is to not make a wiki unusable because I just lost server connection. Is nice to be notified, but once you have loaded a wiki you can operate on it offline. At the end it is a wiki! Can you think in any website that becomes unusable just because you lost connection? 

I think that the module to modify is the sync one, not the tiddlyweb one. The syncadaptor just calls the callback, it's the sync module the one that raises the alerts. 

@pmario my pouchdb plugin stores everything on the local store, so it will never return an error unless something bad happens to the local dB. Sync mechanism is independent and runs in background

**Comment by Jermolene** [comment]
Hi @danielo515 so your concern is to be able to continue to browse the wiki after losing the connection.

I think the best solution might be for the syncer/syncadaptor to be enhanced to track connection status, and to issue an alert just when the connection is first lost.

**Comment by danielo515** [comment]
Hello @Jermolene 

That sounds like the better approach. Regards

**Comment by pmario** [comment]
I just ran across this problem again with the datarchiveadaptor. When the TW wants to write to an archive, where you have no write permission you get all sorts of errors from the different callbacks. The errors can't be suppressed, since they are needed in case of a real errors. 

I had a look at the `logger.alert` function again. It's basically not what I want. 

 - I want to have something like an "audit trail" .. So it's a "log" tiddler, that contains eg: the last 100 messages from the syncer 

 - Those messages should be shown for about 2 seconds at the top of the TW, in the "page - toolbar" area. 

 - after the eg 2 seconds the message should fade away. No movement animations, just fading away.

 - If several messages are shown at the same time, a max number of eg: 3 should be visible at the same time. 

 - The "audit-like-history" should be expandable and permanently visible for testing purpose in the page toolbar area. Or as a sticky "above-story" tiddler. 

Since I need to mess with the syncer anyway. ... datarchiveadaptor, has a "file watch" mechanism, so "syncer polling" can be suspended ... I can prepare a working prototype and create a PR. ... So we can test a bit.

**Comment by pmario** [comment]
#449 may benefit, if this issue is resolved.

**Comment by tobibeer** [comment]
I think we should be able to minimize notifications to a bubble having a counter, whereas when minimized new alerts are only adding to the counter and only when you click the bubble, the notifications are displayed whereas ALL can be dismissed at once.
